### PR TITLE
nixos/dnsmasq: remove deprecated option "extraConfig"

### PR DESCRIPTION
**Pull request by rhoriguchi**
## Description of changes

Remove deprecated option https://github.com/NixOS/nixpkgs/blob/f02fa2f654c7bcc45f0e815c29d093da7f1245b4/nixos/doc/manual/release-notes/rl-2305.section.md?plain=1#L417-L420

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by KoviRobi — APPROVED** [review]
Makes sense to me, I had a quick look at discourse in case anyone had popped up saying some odd configuration can't be represented, wrt deprecation. Also I can't think of any, given it's just command line flags, and we have a way of repeating command line flags (i.e. lists).

I can't merge, and honestly I'm sadly not very active in the NixOS community/development to give good code review, but LGTM.

**Review by fpletz — DISMISSED** [review]
Looks good. But this change needs release notes and a `mkRemovedOptionModule` definition to inform users.

File: nixos/doc/manual/release-notes/rl-2411.section.md [hunk 1]
```diff
@@ -418,6 +418,8 @@
 
 - The `openlens` package got removed, suggested replacment `lens-desktop`
 
+- The `services.dnsmasq.extraConfig` option has been removed, as it had been deprecated for over 2 years. This option is replaced by `services.dnsmasq.settings`.
```
Comment (Scrumplex) [inline nixos/doc/manual/release-notes/rl-2411.section.md:421]: nit: 
```suggestion
- The `services.dnsmasq.extraConfig` option has been removed, as it had been deprecated for over 2 years. This option has been replaced by `services.dnsmasq.settings`.
```

File: nixos/modules/services/networking/dnsmasq.nix [hunk 2]
```diff
@@ -23,20 +23,15 @@ let
...
-  '';
+  dnsmasqConf = settingsFormat.generate "dnsmasq.conf" cfg.settings;
 
 in
 
 {
 
   imports = [
     (mkRenamedOptionModule [ "services" "dnsmasq" "servers" ] [ "services" "dnsmasq" "settings" "server" ])
+    (mkRemovedOptionModule [ "services" "dnsmasq" "extraConfig" ] "This option is replaced by `services.dnsmasq.settings`")
```
Comment (Scrumplex) [inline nixos/modules/services/networking/dnsmasq.nix:34]: nit:
```suggestion
    (mkRemovedOptionModule [ "services" "dnsmasq" "extraConfig" ] "This option has been replaced by `services.dnsmasq.settings`")
```

**Review by Scrumplex — APPROVED** [review]
Apart from the wording, LGTM

**Review by Scrumplex — APPROVED** [review]

**Review by bjornfor — APPROVED** [review]

**Comment by bjornfor** [comment]
> Looks good. But this change needs release notes and a mkRemovedOptionModule definition to inform users.

@fpletz: That's done now.

**Pull request merged**